From: Chad Monroe <chad@monroe.io>
Date: Mon, 24 Nov 2025 14:36:47 -0800
Subject: [PATCH] wifi-scripts: enable EHT beamforming by default

If EHT is supported, always enable EHT beamforming. Emit the EHT
beamforming options from device_htmode_append() in
usr/share/ucode/wifi/hostapd.uc.

Signed-off-by: Chad Monroe <chad@monroe.io>
---
--- a/src/usr/share/ucode/wifi/hostapd.uc
+++ b/src/usr/share/ucode/wifi/hostapd.uc
@@ -451,6 +451,11 @@ function device_htmode_append(config) {
 		config.ieee80211be = true;
 		append_vars(config, [ 'ieee80211be' ]);
 
+		/* force enable EHT beamforming when EHT is supported */
+		append('eht_su_beamformer', 1);
+		append('eht_su_beamformee', 1);
+		append('eht_mu_beamformer', 1);
+
 		if (config.hw_mode == 'a')
 			append_vars(config, [ 'eht_oper_chwidth', 'eht_oper_centr_freq_seg0_idx' ]);
 	}
